python_modules/dagster/dagster_tests/definitions_tests/freshness_checks_tests/test_time_window_partitioned.py

```diff
@@ -0,0 +1,162 @@
+        )
+
+    @multi_asset(
+        outs={
+            "my_partitioned_asset": AssetOut(),
+            "b": AssetOut(),
+        },
+        can_subset=True,
+        partitions_def=DailyPartitionsDefinition(
+            start_date=pendulum.datetime(2020, 1, 1, 0, 0, 0, tz="UTC")
+        ),
+    )
+    def my_multi_asset(context):
+        pass
+
+    result = build_time_window_partitioned_freshness_checks(assets=[my_multi_asset])
+    assert len(result) == 2
+    assert {next(iter(checks_def.check_keys)).asset_key for checks_def in result} == set(
+        my_multi_asset.keys
+    )
+
+    with pytest.raises(Exception, match="Found duplicate assets"):
+        build_time_window_partitioned_freshness_checks(
+            assets=[my_multi_asset, my_multi_asset], offset_minutes=10
+        )
+
+    with pytest.raises(Exception, match="Found duplicate assets"):
+        build_time_window_partitioned_freshness_checks(
+            assets=[my_multi_asset, my_partitioned_asset], offset_minutes=10
+        )
+
+    coercible_key = "blah"
+    result = build_time_window_partitioned_freshness_checks(
+        assets=[coercible_key], offset_minutes=10
+    )
+    assert len(result) == 1
+    assert next(iter(result[0].check_keys)).asset_key == AssetKey.from_coercible(coercible_key)
+
+    with pytest.raises(Exception, match="Found duplicate assets"):
+        build_time_window_partitioned_freshness_checks(
+            assets=[coercible_key, coercible_key], offset_minutes=10
+        )
+
+    regular_asset_key = AssetKey("regular_asset_key")
+    result = build_time_window_partitioned_freshness_checks(
+        assets=[regular_asset_key], offset_minutes=10
+    )
+    assert len(result) == 1
+    assert next(iter(result[0].check_keys)).asset_key == regular_asset_key
+
+    with pytest.raises(Exception, match="Found duplicate assets"):
+        build_time_window_partitioned_freshness_checks(
+            assets=[regular_asset_key, regular_asset_key], offset_minutes=10
+        )
+
+
+def test_result_time_window_partitioned(
+    pendulum_aware_report_dagster_event: None,
+    instance: DagsterInstance,
+) -> None:
+    """Move time forward and backward, with a freshness check parameterized with a cron, and ensure that the check passes and fails as expected."""
+    partitions_def = DailyPartitionsDefinition(
+        start_date=pendulum.datetime(2020, 1, 1, 0, 0, 0, tz="UTC")
+    )
+
+    @asset(partitions_def=partitions_def)
+    def my_asset():
+        pass
+
+    start_time = pendulum.datetime(2021, 1, 1, 1, 0, 0, tz="UTC")
+
+    freshness_checks = build_time_window_partitioned_freshness_checks(
+        assets=[my_asset],
+        offset_minutes=10,
+    )
+
+    freeze_datetime = start_time
+    with pendulum.test(freeze_datetime):
+        # With no events, check fails.
+        assert_check_result(my_asset, instance, freshness_checks, AssetCheckSeverity.WARN, False)
+
+        # Add an event for an old partition. Still fails
```

Comment:
To test that the offset works as expected, I would expect to see an assert that this passes with `pendulum.test(pendulum.datetime(2021, 1, 1, 0, 9, 0, tz="UTC"))`.  I.e. the offset tolerates us being 10 minutes late, so if we're only 9 minutes past the start of the day, then we're fine.